### PR TITLE
[MB-2081] Configure logging levels.

### PR DESCRIPTION
**Pull request by hcrowell**
Configure logging levels for Android and iOS:
* Set default production log level to `error`
* Set default development log level to `debug`
* Optionally set log levels via config.xml:
```
<!-- Urban Airship development log level defaults to debug -->
<preference name="com.urbanairship.development_log_level" value="none | error | warn | info | debug | verbose" />
<!-- Urban Airship production log level defaults to error -->
<preference name="com.urbanairship.production_log_level" value="none | error | warn | info | debug | verbose" />
```

Testing:
* Tested on iPod and nexus5x and verified logs

### CODE REVIEW

File: src/android/CordovaAutopilot.java [hunk 1]
```diff
@@ -224,6 +228,31 @@ public PluginConfig getPluginConfig(Context context) {
     }
 
     /**
+     * Convert the log level string to an int.
+     *
+     * @param logLevel The log level string.
+     * @return The log level.
+     */
+    public int androidLogLevel(String logLevel) {
```
Comment (rlepinski) [inline src/android/CordovaAutopilot.java]: Does it need to be public? Also can we call this `parseLogLevel`

File: src/android/CordovaAutopilot.java [hunk 2]
```diff
@@ -224,6 +228,31 @@ public PluginConfig getPluginConfig(Context context) {
...
+        String logString = logLevel.toLowerCase();
```
Comment (rlepinski) [inline src/android/CordovaAutopilot.java]: probably want to trim it

File: src/android/CordovaAutopilot.java [hunk 3]
```diff
@@ -224,6 +228,31 @@ public PluginConfig getPluginConfig(Context context) {
...
+        if (logString.equals("verbose")) {
```
Comment (rlepinski) [inline src/android/CordovaAutopilot.java]: you can also use equalsIgnoreCase

File: src/android/CordovaAutopilot.java [hunk 4]
```diff
@@ -224,6 +228,31 @@ public PluginConfig getPluginConfig(Context context) {
...
+            return 2;
```
Comment (rlepinski) [inline src/android/CordovaAutopilot.java]: probably return the constants, Log.DEBUG

File: src/ios/UAirshipPlugin.m [hunk 5]
```diff
@@ -908,5 +918,22 @@ - (id)pushEventFromNotification:(UANotificationContent *)notificationContent {
     return result;
 }
 
+-(int)iOSLogLevel:(NSString *)logLevel {
+    NSString *logString = [logLevel lowercaseString];
```
Comment (rlepinski) [inline src/ios/UAirshipPlugin.m]: trim it

File: src/ios/UAirshipPlugin.m [hunk 6]
```diff
@@ -908,5 +918,22 @@ - (id)pushEventFromNotification:(UANotificationContent *)notificationContent {
     return result;
 }
 
+-(int)iOSLogLevel:(NSString *)logLevel {
```
Comment (rlepinski) [inline src/ios/UAirshipPlugin.m]: parseLogLevel

File: src/ios/UAirshipPlugin.m [hunk 7]
```diff
@@ -908,5 +918,22 @@ - (id)pushEventFromNotification:(UANotificationContent *)notificationContent {
...
+    NSString *logString = [logLevel lowercaseString];
+    if ([logString isEqualToString:@"verbose"]) {
+        return 5;
```
Comment (rlepinski) [inline src/ios/UAirshipPlugin.m]: return constants

File: src/ios/UAirshipPlugin.m [hunk 8]
```diff
@@ -908,5 +918,22 @@ - (id)pushEventFromNotification:(UANotificationContent *)notificationContent {
...
+    } else if ([logString isEqualToString:@"info"]) {
+        return 3;
+    } else if ([logString isEqualToString:@"warning"]) {
+        return 2;
+    } else if ([logString isEqualToString:@"error"]) {
+        return 1;
+    } else if ([logString isEqualToString:@"none"]) {
+        return 0;
+    }
+    return 0; // default to none
```
Comment (rlepinski) [inline src/ios/UAirshipPlugin.m]: default to debug for debug, error for release

File: src/android/CordovaAutopilot.java [hunk 9]
```diff
@@ -224,6 +228,31 @@ public PluginConfig getPluginConfig(Context context) {
...
+        } else if (logString.equals("info")) {
+            return 4;
+        } else if (logString.equals("warn")) {
+            return 5;
+        } else if (logString.equals("error")) {
+            return 6;
+        } else if (logString.equals("none")) {
+            return 7;
+        } else {
+            return 7; // default to none
```
Comment (rlepinski) [inline src/android/CordovaAutopilot.java]: default to debug for debug, release for release

File: src/android/CordovaAutopilot.java (hunk 3 above)
Comment (hcrowell) [inline src/android/CordovaAutopilot.java]: When I converted to a switch statement, I got this error:
`platforms/android/src/com/urbanairship/cordova/CordovaAutopilot.java:244: error: strings in switch are not supported in -source 1.6
        switch (logString) {
               ^
  (use -source 7 or higher to enable strings in switch)
1 error
:compileDebugJavaWithJavac FAILED`

**Comment by rlepinski** [comment]
👍

**Comment by madelgi** [comment]
:100:

**Pull request merged**